eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -734,11 +741,16 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
       case Left(cause) =>
         replyTo ! RemoteFailure(cause)
         unlockAndStop(completeTx)
-      case Right((localSpec, localCommitTx, remoteSpec, remoteCommitTx)) =>
+      case Right((localSpec, localCommitTx, remoteSpec, remoteCommitTx, sortedHtlcTxs)) =>
         require(fundingTx.txOut(fundingOutputIndex).publicKeyScript == localCommitTx.input.txOut.publicKeyScript, "pubkey script mismatch!")
         val fundingPubKey = keyManager.fundingPublicKey(channelParams.localParams.fundingKeyPath, purpose.fundingTxIndex)
         val localSigOfRemoteTx = keyManager.sign(remoteCommitTx, fundingPubKey, TxOwner.Remote, channelParams.channelFeatures.commitmentFormat)
-        val localCommitSig = CommitSig(fundingParams.channelId, localSigOfRemoteTx, Nil)
+        val localPerCommitmentPoint = keyManager.htlcPoint(keyManager.keyPath(channelParams.localParams, channelParams.channelConfig))
+        val htlcSignatures = fundingParams.sharedInput_opt match {
+          case Some(_: Multisig2of2Input) => sortedHtlcTxs.map(keyManager.sign(_, localPerCommitmentPoint, purpose.remotePerCommitmentPoint, TxOwner.Remote, channelParams.commitmentFormat)).toList
+          case _ => Nil
+        }
```

Comment:
The pattern matching is unnecessary, when it's not a splice, `sortedHtlcTxs` will be empty.

Reply:
Pulled in [08d7f35](https://github.com/ACINQ/eclair/pull/2720/commits/08d7f352e501d46bed498d6ed468ab838972f2db) unchanged.